data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoClientException.java

```diff
@@ -0,0 +1,13 @@
+package com.microsoft.azure.kusto.data.exceptions;
+
+public abstract class KustoClientException extends Exception {
```

Comment:
DataServiceException inherits from KustoClientException. I assume the motivation is to indicate that this Exception is part of the Java SDK (client). However, when its 2 concrete implementations are either DataServiceException or DataClientException, I think the ambiguous term "Client" can be confusing.

I think it would be better if this class was called DataException (or KustoDataException?).